umap/static/umap/js/modules/sync/updaters.js

```diff
@@ -31,8 +31,8 @@ class BaseUpdater {
     }
   }
 
-  getDataLayerFromID(layerId) {
-    if (layerId) return this.map.getDataLayerByUmapId(layerId)
+  getDataLayerFromID(layerId, future_uuid) {
```

Comment:
Can't we totally replace the layerId by the uuid ?

Reply:
Actually, the layerId **is** an UUID already. Here, we have potentially two different UUIDs. These are from calls in different times:

- A layerId is provided when the updated property of the layer is from a layer that's already saved on the server
- A futureUUID is provided when we're dealing with yet-to-be-saved layers.

I believe we need a way to tell them apart, and we also won't look for them the same way (see `getDataLayerByUmapId`, which is looking using `dl._future_uuid` if it's provided, otherwise `dl.umap_id`)